Can not build detectron2 after adding c++ cocoeval

When building the latest detectron2 (after adding the c++ cocoeval), I met the following error:
```
/data/anaconda3/envs/py374/lib/python3.7/site-packages/torch/include/pybind11/pybind11.h:1133:9:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(const pybind11::detail::initimpl::constructor<Args ...>&, const Extra& ...) [with Args = {long unsigned int, long unsigned int, long unsigned int, double, double, unsigned char, bool}; Extra = {}; type_ = detectron2::COCOeval::InstanceAnnotation; options = {}]’
/data/detectron2/layers/csrc/vision.cpp:121:19:   required from here
/data/anaconda3/envs/py374/lib/python3.7/site-packages/torch/include/pybind11/detail/init.h:63:64: error: no matching function for call to ‘detectron2::COCOeval::InstanceAnnotation::InstanceAnnotation(<brace-enclosed initializer list>)’
 inline Class *construct_or_initialize(Args &&...args) { return new Class{std::forward<Args>(args)...}; }
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /data/detectron2/layers/csrc/vision.cpp:7:0:
/data/detectron2/layers/csrc/cocoeval/cocoeval.h:17:8: note: candidate: detectron2::COCOeval::InstanceAnnotation::InstanceAnnotation()
 struct InstanceAnnotation {
        ^~~~~~~~~~~~~~~~~~
/data/detectron2/layers/csrc/cocoeval/cocoeval.h:17:8: note:   candidate expects 0 arguments, 7 provided
/data/detectron2/layers/csrc/cocoeval/cocoeval.h:17:8: note: candidate: constexpr detectron2::COCOeval::InstanceAnnotation::InstanceAnnotation(const detectron2::COCOeval::InstanceAnnotation&)
/data/detectron2/layers/csrc/cocoeval/cocoeval.h:17:8: note:   candidate expects 1 argument, 7 provided
/data/detectron2/layers/csrc/cocoeval/cocoeval.h:17:8: note: candidate: constexpr detectron2::COCOeval::InstanceAnnotation::InstanceAnnotation(detectron2::COCOeval::InstanceAnnotation&&)
/data/detectron2/layers/csrc/cocoeval/cocoeval.h:17:8: note:   candidate expects 1 argument, 7 provided
error: command 'gcc' failed with exit status 1
```

--- a/detectron2/evaluation/fast_eval_api.py
+++ b/detectron2/evaluation/fast_eval_api.py
@@ -59,12 +59,10 @@ class COCOeval_opt(COCOeval):
             for instance in instances:
                 instance_cpp = _C.InstanceAnnotation(
                     instance["id"],
-                    instance["image_id"],
-                    instance["category_id"],
                     instance["score"] if is_det else instance.get("score", 0.0),
                     instance["area"],
-                    instance.get("iscrowd", 0),
-                    instance.get("ignore", False),
+                    bool(instance.get("iscrowd", 0)),
+                    bool(instance.get("ignore", 0)),
                 )
                 instances_cpp.append(instance_cpp)
             return instances_cpp

--- a/detectron2/layers/csrc/cocoeval/cocoeval.h
+++ b/detectron2/layers/csrc/cocoeval/cocoeval.h
@@ -15,12 +15,17 @@ namespace COCOeval {
 
 // Annotation data for a single object instance in an image
 struct InstanceAnnotation {
-  int64_t id;
-  int64_t image_id;
-  int64_t category_id = 0;
+  InstanceAnnotation(
+      uint64_t id,
+      double score,
+      double area,
+      bool is_crowd,
+      bool ignore)
+      : id{id}, score{score}, area{area}, is_crowd{is_crowd}, ignore{ignore} {}
+  uint64_t id;
   double score = 0.;
   double area = 0.;
-  unsigned char is_crowd = 0;
+  bool is_crowd = false;
   bool ignore = false;
 };
 

--- a/detectron2/layers/csrc/vision.cpp
+++ b/detectron2/layers/csrc/vision.cpp
@@ -111,14 +111,7 @@ PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
       &COCOeval::EvaluateImages,
       "COCOeval::EvaluateImages");
   pybind11::class_<COCOeval::InstanceAnnotation>(m, "InstanceAnnotation")
-      .def(pybind11::init<
-           uint64_t,
-           uint64_t,
-           uint64_t,
-           double,
-           double,
-           unsigned char,
-           bool>());
+      .def(pybind11::init<uint64_t, double, double, bool, bool>());
   pybind11::class_<COCOeval::ImageEvaluation>(m, "ImageEvaluation")
       .def(pybind11::init<>());
 }

